File: ai-ml/lesson-14-gradient-boosting/easy/p03-solve.py
"""
LESSON — Gradient Boosting
EASY P03 — Train gradient boosting with `n_iter_no_change=10` (early stopping). Print the actual number of trees used (`n_estimators_`). Show that it stops before the max.
==================================================

CONCEPT:
  See concepts.md in this lesson folder for detailed explanations.

PROBLEM:
  Train gradient boosting with `n_iter_no_change=10` (early stopping). Print the actual number of trees used (`n_estimators_`). Show that it stops before the max


Write a function `solve()` that implements the solution.

# Check your answer: compare with solutions/p03-solution.py
"""

TRY THIS INPUT:
  Add this test code at the bottom of your file:

  ```python
  X, y = make_classification(n_samples=1000, n_features=20, n_informative=10, n_redundant=5,
  random_state=42)
  X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)

  # Call your function
  result = solve(...)
  print(result)
  ```

EXPECTED OUTPUT:
  Run the solution file to see expected output:
    python3 easy/solutions/p03-solution.py

  Then compare your output format with theirs.

# === WRITE YOUR CODE BELOW ===
# TODO: Write your solution from scratch.

File: html-css-javascript/lesson-16-js-variables-types/medium/p03-solve.js
/**
LESSON 16 — JavaScript Variables & Types
=========================================

PROBLEM: Safe Number Parsing (Medium)
Write parseNumber(str) that converts strings without surprises.

TRY THIS:
  - "" -> 0, "42" -> parseInt, "3.14" -> parseFloat,
    "42px" -> 42 (leading digits), anything else -> NaN.
  - Use regex tests + parseInt/parseFloat/Number appropriately.

EXPECTED OUTPUT:
  parseNumber("42") === 42      parseNumber("3.14") === 3.14
  parseNumber("abc") is NaN     parseNumber("") === 0
  parseNumber("42px") === 42

TEST: node medium/p03-solve.js

CHECK: python3 check.py medium/p03
*/

// TODO: Write your complete solution from scratch below.
